parser_library/src/parsing/parser_impl.cpp

```diff
@@ -14,453 +14,4602 @@
+    parser2 p(this);
+
+    return p.with_model<&parser2::dat_op>(reparse, model_allowed);
 }
 
-bool parser_impl::NOT(const antlr4::Token* token) const
+result_t<semantics::operand_ptr> parser2::alias_op()
 {
-    if (!token)
-        return false;
+    const auto start = cur_pos_adjusted();
+    const auto initial = input.next;
+
+    if (!match<U'C', U'c', U'X', U'x'>(hl_scopes::self_def_type))
+        return failure;
 
-    static constexpr std::string_view NOT_OPERATOR = "NOT";
-    auto token_txt = token->getText();
+    if (!must_follow<U'\''>())
+        return failure;
 
-    return std::ranges::equal(token_txt, NOT_OPERATOR, {}, [](unsigned char c) { return (char)::toupper(c); });
+    auto [error, s] = lex_simple_string();
+    if (error)
+        return failure;
+
+    const auto r = range_from(start);
+    add_hl_symbol(r, hl_scopes::self_def_type);
+
+    return std::make_unique<semantics::expr_assembler_operand>(
+        std::make_unique<expressions::mach_expr_default>(r), capture_text(initial), r);
 }
 
-bool parser_impl::is_attribute_consuming(char c)
+result_t<semantics::operand_ptr> parser2::end_op()
 {
-    auto tmp = std::toupper(static_cast<int>(c));
-    return tmp == 'O' || tmp == 'S' || tmp == 'I' || tmp == 'L' || tmp == 'T';
+    const auto start = cur_pos_adjusted();
+    if (!match<U'('>(hl_scopes::operator_symbol))
+        return failure;
+
+    std::string s1, s2, s3;
+
+    auto op_start = cur_pos_adjusted();
+    while (except<U','>())
+        consume_into(s1);
+    const auto r1 = range_from(op_start);
+
+    if (!match<U','>(hl_scopes::operator_symbol))
+        return failure;
+
+    op_start = cur_pos_adjusted();
+    while (except<U','>())
+        consume_into(s2);
+    const auto r2 = range_from(op_start);
+
+    if (!match<U','>(hl_scopes::operator_symbol))
+        return failure;
+
+    op_start = cur_pos_adjusted();
+    while (except<U')'>())
+        consume_into(s3);
+    const auto r3 = range_from(op_start);
+
+    if (!match<U')'>(hl_scopes::operator_symbol))
+        return failure;
+
+    std::vector<std::unique_ptr<semantics::complex_assembler_operand::component_value_t>> language_triplet;
+    language_triplet.push_back(
+        std::make_unique<semantics::complex_assembler_operand::string_value_t>(std::move(s1), r1));
+    language_triplet.push_back(
+        std::make_unique<semantics::complex_assembler_operand::string_value_t>(std::move(s2), r2));
+    language_triplet.push_back(
+        std::make_unique<semantics::complex_assembler_operand::string_value_t>(std::move(s3), r3));
+    add_hl_symbol(r1, hl_scopes::operand);
+    add_hl_symbol(r2, hl_scopes::operand);
+    add_hl_symbol(r3, hl_scopes::operand);
+
+    const auto r = range_from(start);
+    return std::make_unique<semantics::complex_assembler_operand>("", std::move(language_triplet), r);
 }
 
-bool parser_impl::is_attribute_consuming(const antlr4::Token* token)
+result_t<semantics::operand_ptr> parser2::using_op1()
```

Comment:
```suggestion
result_t<semantics::operand_ptr> parser2::using_op()
```
?

Reply:
This function handles specifically the first operand (which is a bit special).